ssz_serialization/merkleization.nim

```diff
@@ -518,8 +518,7 @@ func bitListHashTreeRoot(merkleizer: var SszMerkleizerImpl, x: BitSeq): Digest =
     if totalBytes == 1:
       # This is an empty bit list.
       # It should be hashed as a tree containing all zeros:
-      return mergeBranches(zeroHashes[merkleizer.topIndex],
-                           zeroHashes[0]) # this is the mixed length
+      return zeroHashes[merkleizer.topIndex]
```

Comment:
Why is this commit considered an improvement? It seems to be that it performs exactly the same number of computations except in the case of an empty list where it acts slightly slower, because it computes the hash of the zero length (which we already know).

Reply:
I'd like to re-use the function in merkle multi-proofs (future code) so that roots can be computed for arbitrary generalized indices (instead of just 1). The `mixInLength` part is only required for generalized index 1 (but not otherwise), so I'd like to push that part out to the caller.